lib/checkPullRequestLabels.js

```diff
@@ -48,27 +136,29 @@ module.exports.checkAssignee = async function(context) {
       // for invalid labels when a pull request is opened.
       return;
     }
-    var labelSubstrings = changelogLabel.split('@');
-    var projectOwner = labelSubstrings[labelSubstrings.length - 1].trim();
 
-    var assignees = [];
-    for (var i = 0; i < pullRequest.assignees.length; i++) {
-      assignees.push(pullRequest.assignees[i].login);
-    }
+    const projectOwner = getProjectOwner(changelogLabel);
 
-    if (
-        projectOwner !== pullRequest.user.login &&
-        !assignees.includes(projectOwner)) {
-      var assigneeParams = context.issue({ assignees: [projectOwner] });
-      await context.github.issues.addAssignees(assigneeParams);
+    const shouldAssignProjectOwner = canAssignProjectOwner(
+      pullRequest, changelogLabel);
+    if (shouldAssignProjectOwner) {
+      const assignees = pullRequest.assignees.map(assignee => assignee.login);
+      if(!assignees.includes(projectOwner)) {
+        var assigneeParams = context.issue({ assignees: [projectOwner] });
+        await context.github.issues.addAssignees(assigneeParams);
 
-      var commentParams = context.issue({
-        body:
-          'Assigning @' +
-          projectOwner +
-          ' for the first-pass review of this pull request. Thanks!',
-      });
-      await context.github.issues.createComment(commentParams);
+        var commentParams = context.issue({
+          body:
+            'Assigning @' +
+            projectOwner +
+            ' for the first-pass review of this pull request. Thanks!',
+        });
+        await context.github.issues.createComment(commentParams);
+      }
+      // Project owner is already assigned so we do nothing.
+    } else {
+      // Attempt to assign all codeowners.
```

Comment:
Would it be better to assign just one codeowner - who owns the max number of files?

\cc @DubeySandeep @seanlip @bansalnitish @srijanreddy98

Reply:
Actually scratch that - checking my comment [here](https://github.com/oppia/oppiabot/issues/75#issuecomment-636035937) - not assigning all codeowners will create the issue of PR with no assignee - we plan to tackle that in M3, so we can update this check then.

Reply:
This is only done when the changelog label is applied, and the changelog label will almost only be applied just after a PR has been created. So I don't think we need a TODO here.

Reply:
Ah I mean - we plan to update the flow later as: assign only one codeowner as the main reviewer and then when a "Needs codeowner approval" label is added, assign the others.

Reply:
I think the `needs code owner` approval label should be for cases where the project owner is one of the reviewers. For these, I don't think we need to create a main reviewer since we might be wrong about it in this case.

Reply:
Hmm ok, let us try this and we can update the checks based on user feedback.